openmc/data/reaction.py

```diff
@@ -0,0 +1,1189 @@
+
+            # Peak ahead to determine type of distribution
+            position = file_obj.tell()
+            params = get_cont_record(file_obj)
+            file_obj.seek(position)
+
+            lang = params[2]
+            if lang == 1:
+                p.distribution = [CorrelatedAngleEnergy.from_endf(file_obj)]
+            elif lang == 2:
+                p.distribution = [KalbachMann.from_endf(file_obj)]
+
+        elif law == 2:
+            # Discrete two-body scattering
+            params, tab2 = get_tab2_record(file_obj)
+            ne = params[5]
+            energy = np.zeros(ne)
+            mu = []
+            for i in range(ne):
+                items, values = get_list_record(file_obj)
+                energy[i] = items[1]
+                lang = items[2]
+                if lang == 0:
+                    mu.append(Legendre(values))
+                elif lang == 12:
+                    mu.append(Tabular(values[::2], values[1::2]))
+                elif lang == 14:
+                    mu.append(Tabular(values[::2], values[1::2],
+                                      'log-linear'))
+
+            angle_dist = AngleDistribution(energy, mu)
+            dist = UncorrelatedAngleEnergy(angle_dist)
+            p.distribution = [dist]
+            # TODO: Add level-inelastic info?
+
+        elif law == 3:
+            # Isotropic discrete emission
+            p.distribution = [UncorrelatedAngleEnergy()]
+            # TODO: Add level-inelastic info?
+
+        elif law == 4:
+            # Discrete two-body recoil
+            pass
+
+        elif law == 5:
+            # Charged particle elastic scattering
+            pass
+
+        elif law == 6:
+            # N-body phase-space distribution
+            p.distribution = [NBodyPhaseSpace.from_endf(file_obj)]
+
+        elif law == 7:
+            # Laboratory energy-angle distribution
+            p.distribution = [LaboratoryAngleEnergy.from_endf(file_obj)]
+
+        products.append(p)
+
+    return products
+
+
+def _get_fission_products_ace(ace):
+    """Generate fission products from an ACE table
+
+    Parameters
+    ----------
+    ace : openmc.data.ace.Table
+        ACE table to read from
+
+    Returns
+    -------
+    products : list of openmc.data.Product
+        Prompt and delayed fission neutrons
+    derived_products : list of openmc.data.Product
+        "Total" fission neutron
+
+    """
+    # No NU block
+    if ace.jxs[2] == 0:
+        return None, None
+
+    products = []
+    derived_products = []
+
+    # Either prompt nu or total nu is given
+    if ace.xss[ace.jxs[2]] > 0:
+        whichnu = 'prompt' if ace.jxs[24] > 0 else 'total'
+
+        neutron = Product('neutron')
+        neutron.emission_mode = whichnu
+
+        idx = ace.jxs[2]
+        LNU = int(ace.xss[idx])
+        if LNU == 1:
+            # Polynomial function form of nu
+            NC = int(ace.xss[idx+1])
+            coefficients = ace.xss[idx+2 : idx+2+NC].copy()
```

Comment:
Why copy here? Setting and accessing list items won't modify the backing list since slicing a list like that creates a new list itself.

```
>>> a = [1,2,3,4,5,6]
>>> b = a[2:4]
>>> b
[3, 4]
>>> b[0] = 100  # Edit this
>>> a
[1, 2, 3, 4, 5, 6]  # `a` is unchanged
>>> 


```

Reply:
`xss` is a numpy array and slicing will only create a view into the memory of the original array